client/container_list.go

```diff
@@ -18,7 +18,7 @@ func (cli *Client) ContainerList(ctx context.Context, options types.ContainerLis
 		query.Set("all", "1")
 	}
 
-	if options.Limit != -1 {
+	if options.Limit != -1 && options.Limit > 0 {
```

Comment:
This could just be one condition now, right? :sweat_smile:

```suggestion
	if options.Limit > 0 {
```

I guess -2, etc should be sent?  That seems a bit strange (to make `-1` special), but I guess that's pre-existing?  There's not a use case for other negative values, right?

Reply:
yeah, so I think "-1" was historically used as "no limit". TBH, if that means that the current cli doesn't work with old daemons (if those would treat 0 as 0), I guess it's already broken.

So, we could make it "if != 0", but that would also send -2 🤔

Reply:
I went for the simple approach as you suggested; just check `> 0` to set the value 👍